### PR TITLE
DATA/assaydev_code_update

### PR DESCRIPTION
**Pull request by vianamp**
Minor changes in my code to adjust to final feature names used in the paper. Also added the notebooks we used to create supplemental figures.

### CODE REVIEW

File: fish_morphology_code/processing/structure_organization/tools/features.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def process_fov(FOVId, df_fov):
         total_area = mask.sum()
 
         # Classification results from CNN classifier
-        classification = data[ch_cla].astype(np.uint8) - 1
+        classification = data[ch_cla].astype(np.uint8)
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/tools/features.py:42]: what does this off by one error fix?

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+    return {
+        "MaxCoeffVar": cvr_corr.max(),
+        "HPeak": hpeak,
+        "PeakDistance": hpeak_dist,
+        "PeakAngle": hpeak_angle,
+    }
+
+
+def ProcessFOV(FOVId, df_fov):
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:192]:
```suggestion
def process_fov(FOVId, df_fov):
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    print(f"Frequency of each level: {np.bincount(input_dig.flatten())}")
+
+    return input_dig
+
+
+def ExpDecay(x, a):
+    return np.exp(-a * x ** 2)
+
+
+def AnalyzeOrientation(
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:54]:
```suggestion
def analyze_orientation(
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    )
+
+    input_dig = np.digitize(input_img, bins) - 1
+
+    print(f"Frequency of each level: {np.bincount(input_dig.flatten())}")
+
+    return input_dig
+
+
+def ExpDecay(x, a):
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:50]:
```suggestion
def exp_decay(x, a):
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    print(f"Bins for percentile normalization: {bins}")
+
+    crop = input_img[y.min() : y.max(), x.min() : x.max()]
+
+    crop[crop > 0] = np.clip(crop[crop > 0], *bins)
+
+    return crop.astype(np.uint16)
+
+
+def QuantizeImage(input_img, nlevels=8):
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:31]:
```suggestion
def quantize_image(input_img, nlevels=8):
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+import lmfit
+import argparse
+import numpy as np
+import pandas as pd
+from skimage import io as skio
+from skimage import feature as skfeature
+from matplotlib import cm, pyplot
+
+
+def GetCroppedCell(input_img, input_msk=None):
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:11]:
```suggestion
def get_cropped_cell(input_img, input_msk=None):
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+    """
+        This combination of parameters display the best performance in ranking
+        10 cells by CV compared to manual ranking.
+    """
+
+    dists = np.linspace(0, dmax, dmax + 1)
+    angles = np.linspace(0, np.pi, nangles)
+
+    crop = GetCroppedCell(raw * (mask == CellLabel))
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:74]:
```suggestion
    crop = get_cropped_cell(raw * (mask == CellLabel))
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+    crop = QuantizeImage(crop, nlevels=nlevels)
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:75]:
```suggestion
    crop = quantize_image(crop, nlevels=nlevels)
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+
+    df_cell["FOVId"] = df_fov.loc[df_cell.RawFileName].FOVId.values
+
+    df_cell = df_cell.set_index(["FOVId", "CellId"])
+
+    df_cell = df_cell.sort_index()
+
+    FOVId = int(args["fov"])
+
+    ProcessFOV(FOVId=FOVId, df_fov=df_cell.loc[(FOVId,)])
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:252]:
```suggestion
    process_fov(FOVId=FOVId, df_fov=df_cell.loc[(FOVId,)])
```

File: fish_morphology_code/processing/structure_organization/figs/orientation.py [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+    filename = os.path.join(source, f"data_output_fov_{FOVId}_bkgrd.tif")
+
+    data = skio.imread(filename)
+
+    mask = skio.imread(df_fov.MaskFileName.values[0])
+
+    df_or = pd.DataFrame()
+    for CellLabel in df_fov.index:
+        s = pd.Series(
+            AnalyzeOrientation(
```
Comment (donovanr) [inline fish_morphology_code/processing/structure_organization/figs/orientation.py:205]:
```suggestion
            analyze_orientation(
```

File: fish_morphology_code/processing/structure_organization/tools/features.py (hunk 1 above)
Comment (vianamp) [reply fish_morphology_code/processing/structure_organization/tools/features.py:42]: To make sure the code run when we have background class that has label 0.

**Pull request merged**